### ISSUE TITLE
myCircuit

### ISSUE THREAD

**Issue by ipekbocegi** [ipekbocegi/myStation#1]
![ir circuit_bb](https://cloud.githubusercontent.com/assets/11627079/8589678/7e530082-2621-11e5-9030-3d9173be7a7d.jpg)

**Comment by z3t0** [comment]
Any particular reason for using two IR leds?

**Comment by ipekbocegi** [comment]
It will be docking station for my robot. I will put a charging connector between the leds. Therefore I want my robot understands the correct position of the connector and connect there precisely. I thought using two leds may provide me the intersection region.

**Comment by z3t0** [comment]
So you're trying to create a rudimentary coordinate system using infra red?

**Comment by khair-eddine** [comment]
@z3t0  i think @ipekbocegi  have the same issue that was in my situation two months ago.

**Comment by z3t0** [comment]
Hmm. I don't think you want to use this library for a coordinate system. Please explain your end goal and I will let you know the best way to reach it. Thanks

**Comment by khair-eddine** [comment]
we both need multiple senders and maby multiple receivers, @ipekbocegi  are you working on swarm robotics ?

**Comment by z3t0** [comment]
But the thing is this library is for sending signals, that is communication. Not for collision detection. What you need is a collision detection system, and then use the library to communicate the info. I can help you with, just please confirm if I'm going in the right direction. Thanks

**Comment by khair-eddine** [comment]
well, i think ipek should answer you personally for that.

**Comment by z3t0** [comment]
Yeah, sorry that was meant to have @ipekbocegi at the top but I was using my phone so I couldnt type that in.

**Comment by ipekbocegi** [comment]
Hi again sorry I was out of town could not answer the posts. 
I have a mobile robot. I want to design automatic docking station for charging the battery. My robot will roughly know the position of charging station in a room, but I need precise docking. So I tought the connector on the robot and the matching one on the station should dock in one trial without collision. I thought if there is 2 leds on station, I can place the connector between them on station. And if I have a receiver on top of robot's connector it can receive the signals and lead robot to the intersection point. Like this study: http://ijcas.org/admin/paper/files/IJCAS_v6_n5_pp.731-739.pdf
![charging](https://cloud.githubusercontent.com/assets/11627079/8794565/6c780ea0-2f8d-11e5-8b43-b4727579d2a6.png)

**Comment by ipekbocegi** [comment]
My plan is two leds transmit different codes like left one sends 11 right one sends 99, receiver checks if it is 99 or not, if valid then writes in an received data array. If coming data is not 99 or 11 accept it as zero and write 0 in that array. After collecting some number of data lets say 20 data, receiver will process that array. It will count the density of the numbers in array. If 99 and 11 are common and there are a few zeros, it will go forward. If there are only 11s or 99s then turn accordingly.

**Comment by z3t0** [comment]
Okay, I see what you are trying to do. But this is not going to work.. at least not in the way you want to do it. the better way to do it would be to have two leds transmit a normal ir signal (not using the library) and then use receivers on the moving part to figure out where it is. This would be more accurate. I;ll draw up a plan once I get the specifics figured out and post it here.

**Comment by z3t0** [comment]
Also you would obviously need collision sensors around your robot to prevent it from crashing into something

**Comment by z3t0** [comment]
This will help: http://letsmakerobots.com/content/how-use-infrared-receiver-sensors-collision-avoidance

**Comment by ipekbocegi** [comment]
Nice tutorial thank you. I think I need IR data not the object detection signal. It has ultrasonic sensors already to detect objects and run away. If I use use IR leds as obj detection it may catch the station but what if it sees other objects? It will try to dock them and move towards them. I need to receive led's emitted data signals and decode it and act on it. 
Actually my codes are working but receiver saves the data in a way I do not want. Therefore now I plan to use two receiver.
I tried using manual TX signals without library but same error occured with receiver. Problem is while receiving two leds, I want receiver array be like: 
99 11 99 11 99 11 99 11 ... 
99 11 99 11 99 11 99 11 ...
...
but it it comes like:
99 99 99 99 99 99 ...
11 11 11 11 11 11 ...
...
I dont understand why..
